assets/src/scripts/components/hydrograph/graph-controls.spec.js

```diff
@@ -0,0 +1,243 @@
+                        }, {
+                            month_nu: '2',
+                            day_nu: '21',
+                            p50_va: '41',
+                            begin_yr: '1970',
+                            end_yr: '2017',
+                            loc_web_ds: 'This method'
+                        }, {
+                            month_nu: '2',
+                            day_nu: '22',
+                            p50_va: '42',
+                            begin_yr: '1970',
+                            end_yr: '2017',
+                            loc_web_ds: 'This method'
+                        }
+                    ]
+                }
+            }
+        },
+        timeSeriesState: {
+            currentVariableID: '45807197',
+            currentDateRange: 'P7D',
+            requestedTimeRange: null,
+            showSeries: {
+                current: true,
+                compare: true,
+                median: true
+            },
+            loadingTSKeys: []
+        },
+        ui: {
+            width: 400
+        }
+    };
+
+    let graphNode;
+
+    beforeEach(() => {
+        let body = select('body');
+        let component = body.append('div')
+            .attr('id', 'hydrograph');
+        component.append('div').attr('class', 'loading-indicator-container');
+        component.append('div').attr('class', 'graph-container');
+        component.append('div').attr('class', 'select-time-series-container');
+        component.append('div').attr('class', 'provisional-data-alert');
+
+        graphNode = document.getElementById('hydrograph');
+    });
+
+    afterEach(() => {
+        select('#hydrograph').remove();
+    });
+
+    describe('graphControls', () => {
+
+        let store;
+        beforeEach(() => {
+            store = configureStore(TEST_STATE);
+            attachToNode(store, graphNode, {siteno: '12345678'});
+        });
+
+        // last year checkbox tests
+        it('Should render the compare toggle checked', () => {
+            const checkbox = select('#last-year-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(true);
+        });
+
+        it('Should render the compare toggle unchecked', () => {
+            store.dispatch(Actions.toggleTimeSeries('compare', false));
+            const checkbox = select('#last-year-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(false);
+        });
+
+        it('should be enabled if there are last year data', () => {
+            expect(select('#last-year-checkbox').property('disabled')).toBeFalsy();
+        });
+
+        it('should be disabled if there are no last year data', () => {
+            store.dispatch(Actions.setCurrentVariable('45807190'));
+            expect(select('#last-year-checkbox').property('disabled')).toBeTruthy();
+        });
+
+        it('Should render one lines', () => {
```

Comment:
These should go back into the index.spec.js test. This does not test that the checkbox is rendered correctly when the state changes. This is testing code in index.js and does not belong here.

Reply:
done!